lib/identity_cache/cache_fetcher.rb

```diff
@@ -11,7 +11,9 @@ def write(key, value)
     end
 
     def delete(key)
-      @cache_backend.write(key, IdentityCache::DELETED, :expires_in => IdentityCache::DELETED_TTL.seconds)
+      result = @cache_backend.write(key, IdentityCache::DELETED, expires_in: IdentityCache::DELETED_TTL.seconds)
+      IdentityCache.logger.debug { "[IdentityCache] delete #{ result ? 'recorded' : 'failed' } for #{key}" }
```

Comment:
The reason you're using blocks here is to avoid allocating the logging string unless you're at the debugging level?